pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -758,3 +726,78 @@ func newUnschedulablePodsMap() *UnschedulablePodsMap {
 		keyFunc: util.GetPodFullName,
 	}
 }
+
+// nominatedPodMap is a structure that stores pods nominated to run on nodes.
+// It exists because nominatedNodeName of pod objects stored in the structure
+// may be different than what scheduler has here. We should be able to find pods
+// by their UID and update/delete them.
+type nominatedPodMap struct {
+	// nominatedPods is a map keyed by a node name and the value is a list of
+	// pods which are nominated to run on the node. These are pods which can be in
+	// the activeQ or unschedulableQ.
+	nominatedPods map[string][]*v1.Pod
+	// nominatedPodToNode is map keyed by a Pod UID to the node name where it is
+	// nominated.
+	nominatedPodToNode map[ktypes.UID]string
+}
+
+func (npm *nominatedPodMap) add(p *v1.Pod, nodeName string) {
+	// always delete the pod if it already exist, to ensure we never store more than
+	// one instance of the pod.
+	npm.delete(p)
+
+	nnn := nodeName
```

Comment:
Can we replace nnn by nominatedNodeName here, the intent will be much clearer.

Reply:
`nnn` is a common pattern in other functions and is more idiomatic in Go than long names.